### PR TITLE
Use --zap-time-encoding flag for log timestamp format

### PR DESCRIPTION
**Pull request by creydr**
**Is this a BUG FIX or a FEATURE ?**:

> Uncomment only one, leave it on its own line:
>
> /kind bug

/kind enhancement

**What this PR does / why we need it**:
Since #843 a more readable format for log timestamps is used. Since v0.11, controller-runtime allows to define the encoding for log timestamps via the `zap-time-encoding` flag (https://github.com/kubernetes-sigs/controller-runtime/pull/1688). This PR makes use of the flag to give the user the possibility to specify the required format (e.g. in case the currently hard coded ISO8601 format is not feasible).

**Special notes for your reviewer**:
To use v0.11 of controller-runtime I had to update the k8s dependencies to 1.23. In addition the go.mod files were adjusted to "use" golang 1.17, which leads to a 2nd `require` block for indirect dependencies (see https://go.dev/doc/go1.17).

Depends-On: #965 and #964 

**Release note**:
```release-note
Use --zap-time-encoding flag to specify the format for timestamp encoding of log messages
```

### CODE REVIEW

File: api/go.mod [hunk 1]
```diff
@@ -1,17 +1,36 @@
...
+	golang.org/x/text v0.3.6 // indirect
+	golang.org/x/tools v0.1.7 // indirect
+	google.golang.org/protobuf v1.27.1 // indirect
+	gopkg.in/check.v1 v1.0.0-20201130134442-10cb98267c6c // indirect
+	gopkg.in/inf.v0 v0.9.1 // indirect
+	gopkg.in/tomb.v1 v1.0.0-20141024135613-dd632973f1e7 // indirect
+	gopkg.in/yaml.v2 v2.4.0 // indirect
+	k8s.io/klog/v2 v2.9.0 // indirect
+	sigs.k8s.io/structured-merge-diff/v4 v4.1.2 // indirect
+)
```
Comment (qinqon) [inline api/go.mod:37]: This wasn't generated with go mod tidy right ?

Comment (creydr) [reply api/go.mod:37]: Actutally I ran a go mod tidy on them

Comment (creydr) [reply api/go.mod:37]: Reran it on api/go.mod without changes

Comment (creydr) [reply api/go.mod:37]: Reran it on ./go.mod, with a minor change (moves go.uber.org/zap to indirect dependencies). Will push it...

**Comment by qinqon** [comment]
/hold

We have to bump prow jobs first with https://github.com/kubevirt/project-infra/pull/1866

**Pull request merged**